### PR TITLE
POC of the rpi package that will handle incoming RFID scans & play so…

### PR DESCRIPTION
**Pull request by itay-sho**
…me audio

### CODE REVIEW

File: 1.raspberry-pi-stuff/octoberry/octoberry/__main__.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    while True:
+        print("Waiting for RFID scans...")
+        rfid_return_value = read_rfid()
+        if not rfid_return_value:
+            continue
+        fc, id = rfid_return_value
+        
+        print("RFID scanned, fd: {fc}, id: {id}")
+
+        response = requests.get(f"{BACKEND_URL}/rfid_check", params={'fc': fc, 'id': id})
```
Comment (itay-sho) [inline 1.raspberry-pi-stuff/octoberry/octoberry/__main__.py:83]: @gregsadetsky API suggestion here, lmk what do you think :)

File: 1.raspberry-pi-stuff/octoberry/octoberry/__main__.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+    if not os.path.exists(cached_sound_path):
+        print("Downloading sound...")
+        r = requests.get(sound_url, allow_redirects=True)
+        open(cached_sound_path, 'wb').write(r.content)
+        print("Sound downloaded")
+
+    print("Playing sound...")
+    mixer = mixer.init()
+    mixer.music.load(cached_sound_path)
```
Comment (itay-sho) [inline 1.raspberry-pi-stuff/octoberry/octoberry/__main__.py:34]: I felt like using pygame here is a little bit weird and prefer something like python-vlc

apparently from all the options suggested in a relevant stack overflow this was according to the comments the most cross-platform and works way

**Comment by itay-sho** [comment]
fixup commits before merge

File: 1.raspberry-pi-stuff/octoberry/octoberry/__main__.py (hunk 1 above)
Comment (gregsadetsky) [reply 1.raspberry-pi-stuff/octoberry/octoberry/__main__.py:83]: this sounds good to start!

File: 1.raspberry-pi-stuff/octoberry/octoberry/__main__.py (hunk 2 above)
Comment (gregsadetsky) [reply 1.raspberry-pi-stuff/octoberry/octoberry/__main__.py:34]: there should be native-ish ways of playing audio...? aplay or mpv..? but it all works.

as long as it supports mp3s and plays well with the raspi audio hardware output, we're good!!

**Comment by gregsadetsky** [comment]
> fixup commits before merge

sorry was this a todo for yourself? just making sure. thanks a ton, amazing work!!!!!!!

**Comment by gregsadetsky** [comment]
and the PR button now says squash and merge --- that will be ok..? we can also pair up to make sure that the repo does the right thing

also just added you as a collaborator to the repo!

**Comment by itay-sho** [comment]
> > fixup commits before merge
> 
> sorry was this a todo for yourself? just making sure. thanks a ton, amazing work!!!!!!!

Yeah it is, and thanks!!

**Comment by itay-sho** [comment]
> and the PR button now says squash and merge --- that will be ok..? we can also pair up to make sure that the repo does the right thing
> 
> also just added you as a collaborator to the repo!

Squash is fine but its auto-generated commit messages can be dirty (especially when having commits with messages like "fix")
There is another git feature called fixup - it is like squash but w/o the message change (similar to -amend)
I think that GitHub doesn't give you the fixup feature as a configuration but you get a prompt to change the message of the squash and potentially removing the "dirty" stuff. So I will probably do that :)

I love good git history, you can build on top of that cool stuff (automatic change logs, etc)
Another git philosophy that I like - each commit in main should be a working state, and then in case of a problem you can always jump one-two commits back and have working conditions. This is also the reason I prefer to squash my commit into one before merging

What do you think?

**Comment by gregsadetsky** [comment]
> What do you think?

I think that - if you don't mind - you could pair and I could see these practices --- because they feel like I get them, but if you explain them to me slowly enough I promise that I will most probably understand haha

but I'm 100000% down with the best practices!! thank you for caring about this!!

**Comment by gregsadetsky** [comment]
just making sure -- this is good to (not) merge? :-) want to discuss?

**Comment by itay-sho** [comment]
> just making sure -- this is good to (not) merge? :-) want to discuss?

Now we can merge :)

**Pull request merged**